### PR TITLE
improve error message to make it actionable

### PR DESCRIPTION
**Pull request by pwntester**
Tiny improvement in error message that helped me debug:

```
Error executing vim.schedule lua callback: Vim:E5555: API call: Wrong type for argument 4, expecting String
```

The error is now showing:

```
Error executing vim.schedule lua callback: Vim:E5555: API call: Wrong type for argument 4 when calling nvim_buf_set_keymap, expecting String
```

### CODE REVIEW

**Comment by teto** [comment]
nice ! tests have to be updated too, you can try locally `make functionaltest TEST_FILE=test/functional/api/keymap_spec.lua`

**Comment by pwntester** [comment]
@teto Fixed some tests, but not sure why remaining failing ones are related with my change. Can yo please take a look?

File: test/functional/api/vim_spec.lua [hunk 1]
```diff
@@ -1512,7 +1512,7 @@ describe('API', function()
   it("does not leak memory on incorrect argument types", function()
     local status, err = pcall(nvim, 'set_current_dir',{'not', 'a', 'dir'})
     eq(false, status)
-    ok(err:match(': Wrong type for argument 1, expecting String') ~= nil)
+    ok(err:match(': Wrong type for argument 1 when calling set_current_dir, expecting String') ~= nil)
```
Comment (teto) [inline test/functional/api/vim_spec.lua:1515]: here it is `nvim_set_current_dir`

Comment (pwntester) [reply test/functional/api/vim_spec.lua]: Thanks! that fixed it, only two more to go. But these two look completely unrelated to this PR to me

**Pull request merged**